compiler/src/dotty/tools/dotc/transform/ExpandPrivate.scala

```diff
@@ -90,8 +90,8 @@ class ExpandPrivate extends MiniPhase with IdentityDenotTransformer { thisPhase
       }
 
       assert(d.symbol.sourceFile != null &&
-             isSimilar(d.symbol.sourceFile.path, ctx.source.file.path),
-          s"private ${d.symbol.showLocated} in ${d.symbol.sourceFile} accessed from ${ctx.owner.showLocated} in ${ctx.source.file}")
+             isSimilar(d.symbol.sourceFile.path, ctx.owner.sourceFile.path),
+          s"private ${d.symbol.showLocated} in ${d.symbol.sourceFile} accessed from ${ctx.owner.showLocated} in ${ctx.owner.sourceFile}")
```

Comment:
When would this make a difference?

Reply:
When unpickling from tasty, `ctx.source.file` will be "Foo.class" whereas `ctx.owner.sourceFile` will be the source file name found in the `SourceFile` annotation of the tasty tree.